### PR TITLE
implemented more extensive acceptance tests than just "include proxysql"

### PR DESCRIPTION
**Pull request by mcrauwel**
#### Pull Request (PR) description

This PR implements more extensive acceptance tests than just "include proxsql"

### CODE REVIEW

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -22,4 +22,177 @@ class { 'proxysql': }
...
+      }
+
+      @proxy_mysql_replication_hostgroup { '10-30':
+        ensure           => 'absent',
+        writer_hostgroup => 10,
+        reader_hostgroup => 30,
+        comment          => 'Test MySQL Cluster 10-30',
+      }
+
+      Proxy_mysql_replication_hostgroup <| |>
```
Comment (bastelfreak) [inline spec/acceptance/class_spec.rb]: I think we don't need to export & collect the resources. That just makes it a bit more complex.

File: spec/acceptance/class_spec.rb [hunk 2]
```diff
@@ -22,4 +22,177 @@ class { 'proxysql': }
       it { is_expected.to be_running }
     end
   end
+
+  context 'extended testing' do
+    # Using puppet_apply as a helper
+    it 'works idempotently with no errors' do
+      pp = <<-EOS
+        class { '::proxysql':
```
Comment (bastelfreak) [inline spec/acceptance/class_spec.rb]: no need for the leading ```::```

File: spec/acceptance/class_spec.rb [hunk 3]
```diff
@@ -22,4 +22,176 @@ class { 'proxysql': }
       it { is_expected.to be_running }
     end
   end
+
+  context 'extended testing' do
+    # Using puppet_apply as a helper
+    it 'works idempotently with no errors' do
+      pp = <<-EOS
+      class { '::proxysql':
```
Comment (bastelfreak) [inline spec/acceptance/class_spec.rb]: no need for the leading ```::```

**Comment by mcrauwel** [comment]
\o/ green!

**Pull request merged**